File: repositories/audit_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type AuditRepository interface {
	Create(ctx context.Context, event *entities.AuditEvent) error
	FindByActorID(ctx context.Context, actorID uuid.UUID) ([]entities.AuditEvent, error)
}

type GormAuditRepository struct {
	db *gorm.DB
}

func NewAuditRepository(db *gorm.DB) AuditRepository {
	return &GormAuditRepository{db: db}
}

func (r *GormAuditRepository) Create(ctx context.Context, event *entities.AuditEvent) error {
	return r.db.WithContext(ctx).Create(event).Error
}

func (r *GormAuditRepository) FindByActorID(ctx context.Context, actorID uuid.UUID) ([]entities.AuditEvent, error) {
	var events []entities.AuditEvent
	err := r.db.WithContext(ctx).Where("actor_id = ?", actorID).Order("timestamp DESC").Find(&events).Error
	return events, err
}

File: python_backend/repositories/audit_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.audit import AuditEvent

class AuditRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, event: AuditEvent) -> None:
        self.db.add(event)
        self.db.commit()
        self.db.refresh(event)

    def FindByActorID(self, actor_id: UUID) -> List[AuditEvent]:
        return self.db.query(AuditEvent).filter(AuditEvent.actor_id == actor_id).order_by(AuditEvent.timestamp.desc()).all()
